plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -0,0 +1,50 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import java.io.File;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspaceRunnable;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+import com.google.common.io.Files;
+
+public class StandardDeployJob implements IWorkspaceRunnable {
```

Comment:
I'm not a fan of this name since it's not actually a Job.  Which is weird since it otherwise looks like a Job.

Why doesn't this just subclass `WorkspaceJob` and call `setRule(project)` in its constructor?  Then you just use normal Eclipse mechanisms to `schedule()` the job.